fw/http_msg.c

```diff
@@ -973,6 +1019,28 @@ tfw_http_msg_del_hbh_hdrs(TfwHttpMsg *hm)
 	return 0;
 }
 
+/**
+ * Remove X-Tempesta-Cache header in the message
+ */
+int
+tfw_http_msg_del_tempesta_cache_hdr(TfwHttpMsg *hm)
+{
+	TfwHttpHdrTbl *ht = hm->h_tbl;
+	unsigned int hid = ht->off;
+	int r = 0;
+
+	do {
+		hid--;
+		if (hid == TFW_HTTP_HDR_X_TEMPESTA_CACHE) {
```

Comment:
This header was made _special_, i.e. it has fixed index in the headers table and you can just check whether the header by the particular index in the table is present or not. Avoiding such loops is the main reason to have special headers.